ophyd/controls/area_detector.py

```diff
@@ -517,17 +518,9 @@ def _insert_fs_resource(self):
                                    {'frame_per_point':
                                     self._num_images.value})
 
-    def _captured_changed(self, value, *args, **kwargs):
-        if value == self._total_images:
-            self._num_captured.clear_sub(self._captured_changed)
-
-            # Close the capture plugin (closes the file)
-
-            self._capture.put(0, wait=True)
-
     def deconfigure(self, *args, **kwargs):
-        self._total_images = self._array_counter.value
-        self._num_captured.subscribe(self._captured_changed)
+        while self._num_captured.value < self._array_counter.value:
+            ttime.sleep(.1)
```

Comment:
Maybe add a logging statement in the while loop that logs every N  seconds with some helpful info? (Ie number if captured values vs number expected.) This could be a perplexing problem if something gets out of sync